iOS-ObjC/MSALiOSobjC/ViewController.m

```diff
@@ -0,0 +1,278 @@
+         
+         - clientId:            The clientID of your application, you should get this from the app portal.
+         - redirectUri:         A redirect URI of your application, you should get this from the app portal.
+         If nil, MSAL will create one by default. i.e./ msauth.<bundleID>://auth
+         - authority:           A URL indicating a directory that MSAL can use to obtain tokens. In Azure AD
+         it is of the form https://<instance/<tenant>, where <instance> is the
+         directory host (e.g. https://login.microsoftonline.com) and <tenant> is a
+         identifier within the directory itself (e.g. a domain associated to the
+         tenant, such as contoso.onmicrosoft.com, or the GUID representing the
+         TenantID property of the directory)
+         - error                The error that occurred creating the application object, if any, if you're
+         not interested in the specific error pass in nil.
+         */
+        
+        NSError *error = nil;
+        
+        NSURL *authorityURL = [NSURL URLWithString:(authorityString)];
+        
+        MSALAuthority *authority = [MSALAuthority authorityWithURL:authorityURL error:nil];
+        
+        
+        MSALPublicClientApplicationConfig *pcaConfig = [[MSALPublicClientApplicationConfig alloc] initWithClientId:clientId
+                                                                                                       redirectUri:NULL
+                                                                                                         authority:authority];
+        
+        application = [[MSALPublicClientApplication alloc] initWithConfiguration:pcaConfig error:&error];
+        
+        if (application == nil)
+        {
+            UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Error initializing application"
+                                                            message:error.description
+                                                           delegate:self
+                                                  cancelButtonTitle:@"OK"
+                                                  otherButtonTitles:nil];
+            [alert show];
+            
+            return;
+        }
+    }
+    
+    - (void)didReceiveMemoryWarning{
+        [super didReceiveMemoryWarning];
+    }
+    
+    /**
+     This action will invoke the remove account APIs to clear the token cache
+     to sign out a user from this application.
+     */
+    
+    - (IBAction)signOut{
+        
+        MSALAccount *account = [self currentAccount];
+        if (account == nil){
+            return;
+        }
+        
+        /**
+         Removes all tokens from the cache for this application for the provided account
+         
+         - account:    The account to remove from the cache
+         */
+        
+        [application removeAccount:account error:nil];
+        _signout.enabled = false;
```

Comment:
please use NO instead. Also move all UI elements to helper methods like in latest Swift sample

Reply:
updated